src/async/search.rs

```diff
@@ -44,40 +44,43 @@ pub fn search_gateway_from(ip: Ipv4Addr, handle: &Handle) -> Box<Future<Item = G
 pub fn search_gateway_from_timeout(
     ip: Ipv4Addr,
     timeout: Duration,
-    handle: &Handle,
-) -> Box<Future<Item = Gateway, Error = SearchError>> {
+) -> Box<Future<Item = Gateway, Error = SearchError> + Send> {
     let addr = SocketAddr::V4(SocketAddrV4::new(ip, 0));
-    let handle = handle.clone();
-    let task = UdpSocket::bind(&addr, &handle)
+    let task = UdpSocket::bind(&addr)
         .into_future()
-        .and_then(|socket| socket.send_dgram(messages::SEARCH_REQUEST.as_bytes(), "239.255.255.250:1900".parse().unwrap()))
+        .and_then(|socket| socket.send_dgram(messages::SEARCH_REQUEST.as_bytes(), &"239.255.255.250:1900".parse().unwrap()))
         .and_then(|(socket, _)| socket.recv_dgram(vec![0u8; 1500]))
         .map_err(|err| SearchError::from(err))
         .and_then(|(_sock, buf, n, _addr)| {
+
             str::from_utf8(&buf[..n])
                 .map_err(|err| SearchError::from(err))
-                .and_then(|text| parsing::parse_search_result(text))
+                .and_then(|text| {
+                    println!("Recv: {:?}", text);
```

Comment:
Forgot a println here.

Reply:
yeah, this library isn't working at all for me atm, so i'm trying to work out why :-/

should be removed, but also, if you have a working environment it'd be great to test.

Reply:
I've tested your branch on my local network and everything seemed to work fine. At what point is an issue occurring? Can it find the gateway? Is it an error while trying to add a port mapping?

The `async` example is kinda tricky because it uses a hard-coded IP address for the local machine which isn't going to work in 99% of the cases. I was planning on rewriting the example so it asks the IP address in the command args.

Reply:
I've [changed](https://github.com/sbstp/rust-igd/commit/15990071edded6999afc5091af2307025e3fc83a) the `async` example to improve it, you could try merging master in your branch and test it out.

Reply:
neat, i'll have a look tonight.

it seems my hue gateway responds first to the SDP request, with a payload that is considered invalid by the parser. have been wondering whether it'd be useful to move to [ssdp](https://crates.io/crates/ssdp) for more general message parsing / encoding.

aside, it's probably also worth renaming the async bits now `async` is a keyword.

Reply:
There's a similar bug that was fixed a few months ago but it wasn't fixed in the async version https://github.com/sbstp/rust-igd/pull/35

Reply:
ahh, that's useful. i re-wrote the async module to handle multiple responses properly and simplifed the logic / API a bit, though still can't seem to convince my network to play ball to test it :-(

Reply:
Does the synchronous version of the library work well?

Reply:
It might also be due to timeouts, I think the default timeout is 3 seconds, so if your router is slower than that it will timeout before its response.

Reply:
ahh it was a gateway misconfiguration, async stuff is working now ^_^

do you have any thoughts on the refactor? only thing that's not implemented is retries, but IMO that could be a separate PR.

Reply:
It's ok if retries aren't implemented just yet. I'll do a new review of the changes tomorrow (because it's 11 PM here) and test it out on my machine. I like what you did with `SearchOptions` and I'd probably add it to the synchronous version (if it's not done already).

You mentioned renaming the async module to something else, I think it makes sense because it will be a reserved keyword really soon. Not sure how to name it exactly, maybe `tokio` or `aio`.